File: src/main/java/mods/eln/sim/thermal/nbt/NbtFurnaceProcess.java
package mods.eln.sim.thermal.nbt;

import mods.eln.misc.INBTTReady;
import mods.eln.sim.thermal.process.FurnaceProcess;
import mods.eln.sim.thermal.ThermalLoad;
import net.minecraft.nbt.NBTTagCompound;

public class NbtFurnaceProcess extends FurnaceProcess implements INBTTReady {

    String name;

    public NbtFurnaceProcess(String name, ThermalLoad load) {
        super(load);
        this.name = name;
    }

    @Override
    public void readFromNBT(NBTTagCompound nbttagcompound, String str) {
        combustibleEnergy = nbttagcompound.getFloat(str + name + "Q");
        setGain(nbttagcompound.getDouble(str + name + "gain"));
    }

    @Override
    public void writeToNBT(NBTTagCompound nbttagcompound, String str) {
        nbttagcompound.setFloat(str + name + "Q", (float) combustibleEnergy);
        nbttagcompound.setDouble(str + name + "gain", getGain());
    }
}

File: src/main/java/mods/eln/sim/nbt/NbtFurnaceProcess.kt
package mods.eln.sim.nbt

import mods.eln.misc.INBTTReady
import mods.eln.sim.FurnaceProcess
import mods.eln.sim.ThermalLoad
import net.minecraft.nbt.NBTTagCompound

class NbtFurnaceProcess(internal var name: String, load: ThermalLoad) : FurnaceProcess(load), INBTTReady {

    override fun readFromNBT(nbttagcompound: NBTTagCompound, str: String) {
        combustibleEnergy = nbttagcompound.getDouble(str + name + "Q")
        gain = nbttagcompound.getDouble(str + name + "gain")
    }

    override fun writeToNBT(nbttagcompound: NBTTagCompound, str: String) {
        nbttagcompound.setDouble(str + name + "Q", combustibleEnergy)
        nbttagcompound.setDouble(str + name + "gain", gain)
    }
}
